File: javascript/problem004.js
var result = 0;

for(i = 100; i< 1000; i++){
  for(j = 100; j < 1000; j++){
    status = true;
    value = i * j;
    str = value.toString();
    for(k = 0; k < str.length; k++){
      if((str.charAt(k) != str.charAt(str.length - 1 - k))){
          status = false;
          break;
      }
    }
    if(status == true && result < value){
      result = value;
    }
  }
}

console.log(result);

File: java/src/Problem004.java
/**
 * Created by 김 on 2016-02-05.
 */
public class Problem004 {

    boolean isPalindrome(String str) {
        for (int k = 0; k < str.length() / 2; k++) {
            if (!str.substring(k, k + 1).equals(str.substring(str.length() - k - 1, str.length() - k))) {
                return false;
            }
        }
        return true;
    }

    int result() {
        int maxValue = 0;
        for (int i = 100; i < 1000; i++) {
            for (int j = 100; j < 1000; j++) {
                int product = i * j;
                String productString = Integer.toString(product);
                if (isPalindrome(productString)) {
                    if (maxValue < product) {
                        maxValue = product;
                    }
                }
            }
        }
        return maxValue;
    }

    public static void main(String args[]) {
        Problem004 problem = new Problem004();
        System.out.println(problem.result());
    }
}